api/api-public-tenant-service-account.go

```diff
@@ -117,6 +120,134 @@ func (s *server) ListServiceAccounts(ctx context.Context, in *pb.ListServiceAcco
 	}, nil
 }
 
+func (s *server) UpdateServiceAccount(ctx context.Context, in *pb.UpdateServiceAccountRequest) (res *pb.InfoServiceAccountResponse, err error) {
+	idRequest := in.GetId()
+	nameRequest := in.GetName()
+	enabledRequest := in.GetEnabled()
+	permisionsIDs := in.GetPermissionIds()
+	if idRequest == "" {
+		return nil, status.New(codes.InvalidArgument, "an id is needed").Err()
+	}
+	if nameRequest == "" {
+		return nil, status.New(codes.InvalidArgument, "an name is needed").Err()
+	}
+	if len(permisionsIDs) == 0 {
+		return nil, status.New(codes.InvalidArgument, "a list of permissions is needed").Err()
+	}
+	// start app context
+	appCtx, err := cluster.NewTenantContextWithGrpcContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		if err != nil {
+			appCtx.Rollback()
+			return
+		}
+	}()
+
+	// Fetch the service account
+	id, err := uuid.FromString(idRequest)
```

Comment:
can you put all this logic inside a re-usable function so I can expose this functionality to the CLI?

Reply:
done

Reply:
`UpdateServiceAccountFields()` was created